doc/src/manual/arrays.md

```diff
@@ -78,6 +78,57 @@ The syntax `[A, B, C, ...]` constructs a 1-d array (vector) of its arguments. If
 arguments have a common [promotion type](@ref conversion-and-promotion) then they get
 converted to that type using `convert`.
 
+For example,
+
+```jldoctest
+julia> bi = Bidiagonal([1, 2., 3+im], [4., 5+im], :U)
+3×3 Bidiagonal{Complex{Float64},Array{Complex{Float64},1}}:
+ 1.0+0.0im  4.0+0.0im      ⋅
+     ⋅      2.0+0.0im  5.0+1.0im
+     ⋅          ⋅      3.0+1.0im
+
+julia> tri = Tridiagonal(ones(2), ones(3), ones(2))
+3×3 Tridiagonal{Float64,Array{Float64,1}}:
+ 1.0  1.0   ⋅
+ 1.0  1.0  1.0
+  ⋅   1.0  1.0
+
+julia> [bi, tri]
+2-element Array{Tridiagonal{Complex{Float64},V} where V<:AbstractArray{Complex{Float64},1},1}:
```

Comment:
I am not exactly sure how promotion here works, but this is not the same as `[promote(bi, tri)...]`? Which produces `Array{Tridiagonal{Complex{Float64},Array{Complex{Float64},1}},1}` (rather than `Array{Tridiagonal{Complex{Float64},V} where V<:AbstractArray{Complex{Float64},1},1}`).

Perhaps nice with an example that promotes "all the way" to a common type?